svn/trunk/SlickEngagement_Plugin.php

```diff
@@ -293,135 +262,204 @@ public function removeSemicolons($value)
-            $boot_data = json_decode($response_text);
-
-            echo "<script>";
-            echo "window.\$slickBoot = window.\$slickBoot || {};";
-            echo "window.\$slickBoot.d = " . $response_text . ";";
-            echo "window.\$slickBoot.s = 'plugin';";
-            echo "window.\$slickBoot._bd = performance.now();";
-            echo "</script>";
-
-            $filmstrip_config = isset($boot_data->filmstrip) ? $boot_data->filmstrip : '';
-            $dcm_config = isset($boot_data->inlineSearch) ? $boot_data->inlineSearch : '';
-            if ( !empty($filmstrip_config) || !empty($dcm_config) ) {
-              $filmstrip_str = empty($filmstrip_config) ? '' :  json_encode($filmstrip_config);
-              $dcm_str = empty($dcm_config) ? '' :  json_encode($dcm_config);
-              echo "<script>\n";
-              echo "/* Slickstream CLS Insertion */\n";
-              echo '"use strict";(async(e,t)=>{const n=e?JSON.parse(e):null;const r=t?JSON.parse(t):null;if(n||r){const e=async()=>{if(document.body){if(n){o(n.selector,n.position||"after selector","slick-film-strip",n.minHeight||72)}if(r){r.forEach((e=>{if(e.selector){o(e.selector,e.position||"after selector","slick-inline-search-panel",e.minHeight||350,e.id)}}))}return}window.requestAnimationFrame(e)};window.requestAnimationFrame(e)}const c=async(e,t)=>{const n=Date.now();while(true){const r=document.querySelector(e);if(r){return r}const c=Date.now();if(c-n>=t){throw new Error("Timeout")}await i(200)}};const i=async e=>new Promise((t=>{setTimeout(t,e)}));const o=async(e,t,n,r,i)=>{try{const o=await c(e,5e3);const s=i?document.querySelector(`.${n}[data-config="${i}"]`):document.querySelector(`.${n}`);if(o&&!s){const e=document.createElement("div");e.style.minHeight=r+"px";e.classList.add(n);if(i){e.dataset.config=i}switch(t){case"after selector":o.insertAdjacentElement("afterend",e);break;case"before selector":o.insertAdjacentElement("beforebegin",e);break;case"first child of selector":o.insertAdjacentElement("afterbegin",e);break;case"last child of selector":o.insertAdjacentElement("beforeend",e);break}return e}}catch(t){console.log("plugin","error",`Failed to inject ${n} for selector ${e}`)}return false}})' . "\n";
-              echo "('" . addslashes($filmstrip_str) . "','" . addslashes($dcm_str) . "');" . "\n";
-              echo "</script>\n";
+        $headers = array('referer' => home_url());
+        $response = wp_remote_get($remote , array('timeout' => 2, 'headers' => $headers));
+        
+        if (is_array($response)) {
+            $response_text = wp_remote_retrieve_body($response);
+            return json_decode($response_text);
+        } else {
+            return null;
+        }
+    }
+
+    private function echoSlickBootJs($boot_data_obj) {
+      echo <<<JSBLOCK
+        <script>
+        window.\$slickBoot = window.\$slickBoot || {};
+        window.\$slickBoot.d = ' . json_encode($boot_data_obj) . ';
```

Comment:
Fatal error: Uncaught Error: Object of class stdClass could not be converted to string
in /nas/content/live/slicksampleprd/wp-content/plugins/slick-engagement/SlickEngagement_Plugin.php on line 287